tests/ViewTest.php

```diff
@@ -142,6 +142,28 @@ private function createFileStructure(array $items, string $baseDirectory = null)
         }
     }
 
+    public function testViewFileNullBeginPage(): void
+    {
+        $config = require Builder::path('tests');
+
+        $container = new Container($config);
+        $view = $container->get(View::class);
+
+        $this->expectException(\LogicException::class);
+        $view->beginPage();
+    }
+
+    public function testViewFileNullEndPage(): void
+    {
+        $config = require Builder::path('tests');
+
+        $container = new Container($config);
+        $view = $container->get(View::class);
+
+        $this->expectException(\LogicException::class);
+        $view->endPage();
```

Comment:
I think `endPage()` should fall down if `beginPage()` was not called before.
So this test case is wrong.

Reply:
In this case we should test for exception message string.